contracts/strategies/direct-grants-simple/DirectGrantsSimpleStrategy.sol

```diff
@@ -0,0 +1,388 @@
+    }
+
+    /// @notice Reject pending milestone of the recipient
+    /// @param _recipientId Id of the recipient
+    /// @param _milestoneId Id of the milestone
+    function rejectMilestone(address _recipientId, uint256 _milestoneId) external onlyPoolManager(msg.sender) {
+        Milestone storage milestone = milestones[_recipientId][_milestoneId];
+        if (milestone.milestoneStatus == RecipientStatus.Accepted) {
+            revert MILESTONE_ALREADY_ACCEPTED();
+        }
+
+        milestone.milestoneStatus = RecipientStatus.Rejected;
+        emit MILESTONE_REJECTED(_recipientId, _milestoneId);
+    }
+
+    /// @notice Set the internal status of the recipient to InReview
+    /// @param _recipientIds Ids of the recipients
+    function setIntenalRecipientStatusToInReview(address[] calldata _recipientIds)
+        external
+        onlyPoolManager(msg.sender)
+    {
+        uint256 recipientLength = _recipientIds.length;
+        for (uint256 i = 0; i < recipientLength;) {
+            _recipients[_recipientIds[i]].recipientStatus = InternalRecipientStatus.InReview;
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice Withdraw funds from pool
+    /// @param _amount The amount to be withdrawn
+    function withdraw(uint256 _amount) external onlyPoolManager(msg.sender) {
+        allo.decreasePoolTotalFunding(poolId, _amount);
+        _transferAmount(allo.getPool(poolId).token, msg.sender, _amount);
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @notice Register to the pool
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the transaction
+    function _registerRecipient(bytes memory _data, address _sender)
+        internal
+        override
+        onlyActivePool
+        returns (address recipientId)
+    {
+        address recipientAddress;
+        bool isRegistryIdentity;
+        uint256 grantAmount;
+        Metadata memory metadata;
+
+        // decode data custom to this strategy
+        if (registryGating) {
+            (recipientId, recipientAddress, grantAmount, metadata) =
+                abi.decode(_data, (address, address, uint256, Metadata));
+
+            if (!_isIdentityMember(recipientId, _sender)) {
+                revert UNAUTHORIZED();
+            }
+        } else {
+            (recipientAddress, isRegistryIdentity, grantAmount, metadata) =
+                abi.decode(_data, (address, bool, uint256, Metadata));
+            recipientId = _sender;
+            if (isRegistryIdentity && !_isIdentityMember(recipientId, _sender)) {
+                revert UNAUTHORIZED();
+            }
+        }
+
+        if (grantAmountRequired && grantAmount == 0) {
+            revert INVALID_REGISTRATION();
+        }
+        if (upcomingMilestone[recipientId] != 0) {
```

Comment:
we could introduce a check here to check if a recipientId has already the status accepted. Since a recipient could already be accepted and has no milestone yet